demisto_sdk/commands/update_release_notes/update_rn.py

```diff
@@ -80,55 +110,371 @@ def get_deprecated_comment_from_desc(description: str) -> str:
         )
-        rn_from_description = get_deprecated_comment_from_desc(description)
-        return f'- Deprecated. {rn_from_description or "Use %%% instead"}.\n'
+    return rn
 
-    if file_type != FileType.INTEGRATION:
-        return ""
 
-    # look for deprecated commands
+def generate_deprecation_rn(
+    name: str,
+    new_content: Union[Parameter, Argument, Command],
+    old_content: Union[Parameter, Argument, Command],
+    content_type: content_type,
+    parent: Union[str, None] = None,
+) -> str:
+    """Generates release notes for deprecated content items.
+
+    Args:
+        name (str): The name of the content item being deprecated.
+        old_content (Union[Parameter, Argument, Command]): The previous version of the content item.
+        new_content (Union[Parameter, Argument, Command]): The updated version of the content item.
+        content_type (content_type): The type of the content item (e.g., argument, command, parameter).
+        parent (Union[str, None], optional): The name of the parent content item, if applicable
+                                             (e.g., the command name for a deprecated argument). Defaults to None.
+
+    Returns:
+        str: A formatted release note indicating the deprecation of the item. Returns an empty string if no deprecation is detected.
+    """
+    if new_content.deprecated and not old_content.deprecated:
+        if isinstance(new_content, Argument) and parent:
+            return DEPRECATED_ARGUMENT.format(
+                name=name, type=content_type.value, command_name=parent
+            )
+        return GENERAL_DEPRECATED_RN.format(
+            name=name,
+            type=content_type.value,
+            replacement=(
+                get_deprecated_comment_from_desc(new_content.details_for_rn())
+                or "Use %%% instead"
+            ),
+        )
+    return ""
+
+
+def generate_required_rn(
+    name: str,
+    new_content: Union[Parameter, Argument, Command],
+    old_content: Union[Parameter, Argument, Command],
+    content_type: content_type,
+) -> str:
+    """Generates release notes for content items that have been updated to be required.
+
+    Args:
+        name (str): The name of the content item being updated.
+        new_content (Union[Parameter, Argument, Command]): The updated version of the content item.
+        old_content (Union[Parameter, Argument, Command]): The previous version of the content item.
+        content_type (content_type): The type of the content item (e.g., parameter, argument, command).
+
+    Returns:
+        str: A formatted release note indicating that the specified content item is now required.
+             Returns an empty string if no changes to the "required" status are detected.
+    """
+    if (
+        (
+            (isinstance(new_content, Parameter) and isinstance(old_content, Parameter))
+            or (isinstance(new_content, Argument) and isinstance(old_content, Argument))
```

Comment:
why you need to check both new and old content?

Reply:
just in order to not use #ignore